jenkins/Jenkinsfile

```diff
@@ -0,0 +1,148 @@
+def secrets = [
+  [path: 'secret/jenkins/github', engineVersion: 2, secretValues: [
+    [envVar: 'PRIVATE_REPO_GITHUB_TOKEN', vaultKey: 'private_token']]],
+]
+
+def configuration = [vaultUrl: "${env.VAULT_URL}",  vaultCredentialId: 'vault-app-role', engineVersion: 2]
+
+def getClusterManagerVersion(String branchName='master'){
+  sh returnStdout:true, script: """
+    eval "\$(curl 'https://raw.githubusercontent.com/cloudify-cosmo/cloudify-cluster-manager/${branchName}/packaging/version_info')"
+    echo "\$CLOUDIFY_CLUSTER_MANAGER_VERSION"
+  """
+}
+
+def getClusterManagerPreRelease(String branchName='master'){
+  sh returnStdout:true, script: """
+    eval "\$(curl 'https://raw.githubusercontent.com/cloudify-cosmo/cloudify-cluster-manager/${branchName}/packaging/version_info')"
+    echo "\$CLOUDIFY_CLUSTER_MANAGER_PACKAGE_RELEASE"
+  """
+}
+
+@Library('pipeline-shared-library') _
+
+pipeline {
+  agent {
+    kubernetes {
+      label 'cluster-manager'
+      defaultContainer 'jnlp'
+      yamlFile 'jenkins/build-pod.yaml'
+    }
+  }
+
+  options {
+    checkoutToSubdirectory('cloudify-cluster-manager')
+    buildDiscarder(logRotator(numToKeepStr:'30'))
```

Comment:
can you put 10 instead? trying to clean unneeded logs as much as possible

Reply:
Sure, changed.